falcon_generate_changelist.py

```diff
@@ -0,0 +1,132 @@
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Creates CHANGES.txt from git history.
+# "Inspired by" https://github.com/apache/spark/blob/master/dev/create-release/generate-changelist.py
+#
+# Usage:
+#   First set the new release version and previous release Git tag against which the changelog needs to be generated.
+#   $  python generate-changelist.py
+
+
+import os
+import sys
+import subprocess
+import time
+import traceback
+import re
+import glob
+
+## Set the below variables before generating the CHANGES
+NEW_RELEASE_VERSION = "0.10-SNAPSHOT"
+PREV_RELEASE_GIT_TAG = "release-0.9-rc0"
+CAPITALIZED_PROJECT_NAME = "falcon".upper()
+
+CHANGELIST = "CHANGES.txt"
+OLD_CHANGELIST = "%s.old" % (CHANGELIST)
+NEW_CHANGELIST = "%s.new" % (CHANGELIST)
+TMP_CHANGELIST = "%s.tmp" % (CHANGELIST)
+
+LOG_FILE_NAME = "changes_%s" % time.strftime("%h_%m_%Y_%I_%M_%S")
+LOG_FILE = open(LOG_FILE_NAME, 'w')
+
+
+def run_cmd(cmd):
+    try:
+        print >> LOG_FILE, "Running command: %s" % cmd
+        output = subprocess.check_output(cmd, shell=True, stderr=LOG_FILE)
+        print >> LOG_FILE, "Output: %s" % output
+        return output
+    except:
+        traceback.print_exc()
+        cleanup()
+        sys.exit(1)
+
+
+def append_to_changelist(file_name, string):
+    with open(file_name, "a") as f:
+        print >> f, string
+
+
+def cleanup(ask=True):
+    if ask is True:
+        print "OK to delete temporary and log files? (y/N): "
+        response = raw_input()
+    if ask is False or (ask is True and response == "y"):
+        tmp_files = glob.glob(os.path.join(TMP_CHANGELIST + "*"))
+        for f in tmp_files:
+            os.remove(f)
+        if os.path.isfile(OLD_CHANGELIST):
+            os.remove(OLD_CHANGELIST)
+        LOG_FILE.close()
+        os.remove(LOG_FILE_NAME)
+
+
+print "Generating new %s for FALCON release %s" % (CHANGELIST, NEW_RELEASE_VERSION)
+if os.path.isfile(TMP_CHANGELIST):
+    os.remove(TMP_CHANGELIST)
+if os.path.isfile(OLD_CHANGELIST):
+    os.remove(OLD_CHANGELIST)
+
+print "Getting commits between tag %s and HEAD" % PREV_RELEASE_GIT_TAG
+hashes = run_cmd("git log %s..HEAD --pretty='%%h'" % PREV_RELEASE_GIT_TAG).split()
+
+print "Getting details of %s commits" % len(hashes)
+
+for h in hashes:
```

Comment:
While some projects may be generating change log like (putting in commit order instead of making different sections) I really prefer the way we generate the change log over the other approach. 

I would also like to add one improvement to it on the lines of Hadoop project. We should put all sub tasks under the parent task as they follow same category, this adds valuable context to the change log and makes it more readable.